packages/enzyme-adapter-utils/src/Utils.js

```diff
@@ -266,19 +266,35 @@ export function simulateError(
   hierarchy,
   getNodeType = nodeTypeFromType,
   getDisplayName = displayNameOfNode,
+  catchingType = {},
 ) {
-  const { componentDidCatch } = catchingInstance || {};
-  if (!componentDidCatch) {
+  const instance = catchingInstance || {};
+
+  const { componentDidCatch } = instance;
+
+  const { getDerivedStateFromError } = catchingType;
+
+  if (!componentDidCatch && !getDerivedStateFromError) {
```

Comment:
I think we should also pass in a boolean, `checkForGetDerivedStateFromError` or something - because in react < 16.6, that logic shouldn't apply.